src/main/java/seedu/address/storage/JsonAdaptedPicture.java

```diff
@@ -30,14 +35,27 @@ public JsonAdaptedPicture(Picture source) {
         filePath = source.getFilePath();
     }
 
+    private IllegalValueException internalIllegalValueException(String message) {
+        logger.warning(String.format(MESSAGE_DESERIALIZE_ERROR_DUMP_DATA, "Picture}"));
+        logger.warning(this.toString());
+        return new IllegalValueException(message);
+    }
+
     /**
      * Converts this Jackson-friendly adapted picture object into the model's {@code Picture} object.
      */
     public Picture toModelType() throws IllegalValueException {
         if (!Picture.isValidFilePath(filePath)) {
```

Comment:
I think we can rename `Picture#isValidFilePath` to `isValid` or something since it checks more than just the path